File: DSA-DAILY/27-07-2025/2210. Count Hills and Valleys in an Array@leetcode-easy.java
import java.util.*;
class Solution {
    public int countHillValley(int[] nums) {
        int res=0,j=0,n=nums.length;
        for(int i=1;i<n-1;i++){
            if((nums[j]<nums[i]&&nums[i]>nums[i+1])||(nums[j]>nums[i]&&nums[i]<nums[i+1])){
                res++;
                j=i;
            }
        }
        return res;
    }

    public static void main(String[] args) {
        Solution solution=new Solution();
       // int[] nums = {2, 4, 1, 1, 6, 5}; output: 3
       int n;
       Scanner scanner = new Scanner(System.in);
       System.out.print("Enter the number of elements in the array: ");
       n = scanner.nextInt();
       int[] nums = new int[n];
       System.out.println("Enter the elements of the array:");
       for (int i = 0; i < n; i++) {
           nums[i] = scanner.nextInt();
       }
        int result = solution.countHillValley(nums);
        System.out.println("Number of hills and valleys: " + result); 
    }
}

File: DSA-DAILY/DSA-DAILY/27-07-2025/2210. Count Hills and Valleys in an Array@leetcode-easy.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countHillValley = function(nums) {
    let res=0,j=0,n=nums.length
    for(let i=1;i<n-1;i++){
        if((nums[j]<nums[i] && nums[i]>nums[i+1])||(nums[j]>nums[i]&&nums[i]<nums[i+1])){
            res++;
            j=i;
        }
    }
    return res;
};

const readline=require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
})

readline.question('Enter the array elements seperated by space:', input=>{
        const nums=input.trim().split(/\s+/).map(Number);
        const result=countHillValley(nums);
        console.log("Input array: ",nums);
        console.log("Number of hills and valleys: ",result);
        readline.close();
}) 